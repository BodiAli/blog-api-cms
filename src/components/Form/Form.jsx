import PropTypes from "prop-types";
import logBlogIcon from "/images/the-log-blog-icon.svg";
import styles from "./Form.module.css";

export default function Form({ children, onSubmit }) {
  return (
    <form className={styles.form} onSubmit={onSubmit}>
      <h1>
        <img src={logBlogIcon} alt="The Log Blog icon" />
        The Log Blog
      </h1>
      {children}
    </form>
  );
}

Form.propTypes = {
  children: PropTypes.element.isRequired,
  onSubmit: PropTypes.func.isRequired,
};
