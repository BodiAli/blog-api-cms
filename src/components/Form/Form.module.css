.form {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 1rem;
  padding: 1rem;
}

@media (max-width: 800px) {
  .form {
    padding: 1rem 0;
  }
}
