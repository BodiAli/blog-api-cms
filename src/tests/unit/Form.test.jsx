import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Form from "../../components/Form/Form";

describe("Form component", () => {
  test("should render a form", () => {
    const { container } = render(
      <Form>
        <p>Form</p>
      </Form>
    );

    expect(container).toMatchSnapshot();
  });

  test("should render children passed to the Form component", () => {
    render(
      <Form>
        <h1>heading 1</h1>
        <h2>heading 2</h2>
      </Form>
    );

    expect(screen.getByRole("heading", { level: 1, name: "heading 1" })).toBeInTheDocument();
    expect(screen.getByRole("heading", { level: 2, name: "heading 2" })).toBeInTheDocument();
  });

  test("should call onSubmit prop when form is submitted", async () => {
    const user = userEvent.setup();
    const handleSubmit = vi.fn((e) => {
      e.preventDefault();
    });

    render(
      <Form onSubmit={handleSubmit}>
        <button type="submit">Submit</button>
      </Form>
    );

    const submitButton = screen.getByRole("button", { name: "Submit" });

    await user.click(submitButton);

    expect(handleSubmit).toHaveBeenCalledOnce();
  });
});
