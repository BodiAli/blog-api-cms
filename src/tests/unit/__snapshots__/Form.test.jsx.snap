// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Form component > should render a form 1`] = `
<div>
  <form
    class="_form_440ad4"
  >
    <p>
      Form
    </p>
  </form>
</div>
`;
